packages/client/src/shim/node-fetch.ts

```diff
@@ -1,7 +1,7 @@
 type FetchResponse = Response
 // In browsers, the node-fetch package is replaced with this to use native fetch
 export default (
-    ((typeof fetch !== 'undefined' && fetch) || (typeof window !== 'undefined' && window.fetch) || undefined)!
+    ((typeof window !== 'undefined' && window.fetch.bind(window)) || (typeof fetch !== 'undefined' && fetch) || undefined)!
```

Comment:
Is the order change necessary? I'd imagine this should work just fine and make for a smaller diff?
```js
((typeof fetch !== 'undefined' && fetch) || (typeof window !== 'undefined' && window.fetch.bind(window)) || undefined)!
```
We may also want to proactively cover a situation in which `window.fetch` is `undefined`, meaning `window.fetch.bind` will explode if that happens.

Other than that it's a very good change!

Reply:
Unfortunately, it is necessary, in our case both fetch and window.fetch had values and if I didn't change the orders same error will raise.

I'm not sure that scenario will happen but I'll add it in a second.

Reply:
Ah, isn't `fetch` equal to `window.fetch` in the browser? That'd make sense and changing the order is the right thing to do, in the end.

About that extra scenario, right, most `window` objects (96%+) will have `fetch` on them. Hm, I'm fine with both (with and without the extra check).

Reply:
> Ah, isn't `fetch` equal to `window.fetch` in the browser?

They are the same in the browser. If I want to keep the order (Like below) I had to add a new check in the first condition to be sure window is not undefined. So for keeping the conditions simple I changed the orders. 

```
((typeof fetch !== 'undefined' && typeof window !== 'undefined' && fetch.bind(window)) || (typeof window !== 'undefined' && window.fetch.bind(window)) || undefined)!
```